app/manager/models.py

```diff
@@ -18,163 +19,126 @@ def __str__(self):
+    environment = models.ForeignKey(Environment, null=False, blank=False, on_delete=models.PROTECT)
 
     def __str__(self):
-        """String for representing"""
-        return self.name
+        return self.room.__str__() + " : " + self.environment.__str__() + " : " + self.name
 
+    class Meta:
+        unique_together = [['room', 'name', 'environment']]
 
-class Server(models.Model):
-    """Model class for defining a server"""
 
-    # Fields
-    name = models.CharField(max_length=64, null=False)
-    cpu = models.CharField(max_length=64, null=False)
-    memory = models.PositiveIntegerField()
-    u_size = models.PositiveSmallIntegerField(default=1)
-    end_support = models.DateField()
-
-    # Metadata
-    class Meta: 
-        # Default ordering
-        ordering = ['name']
+class HostType(models.Model):
+    name = models.CharField(max_length=64, null=False, blank=False, unique=True)
 
     def __str__(self):
-        """String for representing"""
         return self.name
 
 
-class Cluster(models.Model):
-    """Model class for defining a cluster"""
+class HostStatus(models.Model):
+    name = models.CharField(max_length=64, null=False, blank=False, unique=True)
 
-    # Fields
-    name = models.CharField(max_length=64, null=False)
-    env = models.ForeignKey(EnvType, on_delete=models.CASCADE)
+    def __str__(self):
+        return self.name
 
-    # Metadata
-    class Meta: 
-        # Default ordering
-        ordering = ['name']
+
+class Service(models.Model):
+    name = models.CharField(max_length=64, null=False, blank=False, unique=True)
 
     def __str__(self):
-        """String for representing"""
         return self.name
 
 
 class Group(models.Model):
-    """Model class for defining groups"""
-
-    # Fields
-    name = models.CharField(max_length=64, null=False)
-    cluster = models.ForeignKey(Cluster, on_delete=models.CASCADE)
-
-    # Metadata
-    class Meta: 
-        # Default ordering
-        ordering = ['name']
+    name = models.CharField(max_length=64, null=False, blank=False, unique=True)
 
     def __str__(self):
-        """String for representing"""
         return self.name
 
 
 class Host(models.Model):
-    """Model class for defining a host"""
+    hostname = models.CharField(max_length=128, null=False, blank=False, unique=True)
+    fqdn_hostname = models.CharField(max_length=128, null=False, blank=False, unique=True)
+    generation = models.CharField(max_length=64, null=True, blank=True)
+    services = models.ManyToManyField(Service)
+    groups = models.ManyToManyField(Group)
+    cpu_nb = models.IntegerField(null=False, blank=False)
+    cpu_speed = models.FloatField(null=False, blank=False)
+    cores_nb = models.IntegerField(null=False, blank=False)
+    threads_nb = models.IntegerField(null=False, blank=False)
+    gpu_speed = models.FloatField(null=False, blank=False)
+    gpu_nb = models.IntegerField(null=False, blank=False)
```

Comment:
Seeing these fields here give me the idea of doing a Model for the hardware like for the address or disk that we could call HostHardware, adding generation field too

Reply:
I don't really see the need since what I understand is that we will modify the hardware configuration, not add a new one to the existing one.

Reply:
For the hardware part I saw it like the VM profile, we could have a class/table where we have the different type of model like VM small, medium, large, xlarge, not focused on the name but on cpu,ram,gpu, no need to duplicate these data in the Host Class, just need a foreign key